tests/test_interval_index.py

```diff
@@ -31,6 +31,11 @@ def test_from_tuples() -> None:
     )
 
 
+def test_to_tuples() -> None:
+    ind = pd.IntervalIndex.from_tuples([(0, 1), (1, 2)]).to_tuples()
+    check(assert_type(ind, pd.Index), pd.Index)
```

Comment:
Can you change this test to 
```python
check(assert_type(ind, pd.Index), pd.Index, tuple)
```

That will then check we get an `Index` of tuples.

Reply:
Actually I wrote such a test at first, but finally removed the last `tuple` argument because I think it is not necessary to check the type of elements in the `Index` in this test at this point.

If Indexes are generic now, then this test should be like this:

```py
check(assert_type(ind, "pd.Index[tuple]"), pd.Index, tuple)
```

in which case it is important to check that the inferred return type is *actually* an instance of an Index of tuples.

However, since Indexes are not generic at the moment, it is not the responsibility of this stub to ensure that a return value is actually an Index of tuples.
For now, this type stub layer should only guarantee the behavior of "returning an Index (of something)".

So, I think the current test is fine for now, and checking for the type of elements in an Index will make sense after generic support for indexes is added in the future (at that time this test should be modified to the code above).

Are you suggesting that this test should check for the type of elements in an Index even though it is currently a test for runtime behavior and not a test for this type stub?

Reply:
>  I think the current test is fine for now, and checking for the type of elements in an Index will only make sense after generic support for Indexes is added in the future (at that time this test should be modified to the code above).
> 
> Are you suggesting that this test should check for the type of elements in the Index even though it is currently a test for runtime behavior and not a test for this type stub?

The `assert_type()` call checks the validity of the stub.  The `check()` call validates that the stub and runtime behavior are aligned.  By adding the `tuple` argument to `check`, it is just making sure that alignment is correct.  

We've done that (admittedly inconsistently) in other places, and it actually has revealed some errors in pandas itself, or in the stubs.  So the goal is to try to get more consistency in how we do our testing.

I agree that things will be better once we make `Index` generic in the stubs.